addons/spreadsheet/static/src/data_sources/display_name_repository.js

```diff
@@ -118,85 +96,35 @@ export class DisplayNameRepository {
-                whenDataIsFetched: () => this.dataFetchedCallback(),
-                successCallback: this._assignResult.bind(this),
-                failureCallback: this._assignError.bind(this),
-            });
-        }
-        return this._endpoints[model];
-    }
-
-    /**
-     * This method is called when the display name of a record is successfully
-     * fetched. It updates the cache and resolves the deferred of the record.
-     *
-     * @param {Request} request
-     * @param {BatchedNameGetRPCResult} result
-     *
-     * @private
-     */
-    _assignResult(request, result) {
-        const deferred = this._displayNames[request.resModel][request.args[0]].deferred;
-        deferred.resolve(result && result[1]);
-        this._displayNames[request.resModel][request.args[0]] = {
-            state: "COMPLETED",
-            deferred,
-            value: result && result[1],
-        };
-    }
-
-    /**
-     * This method is called when the display name of a record could not be
-     * fetched. It updates the cache and rejects the deferred of the record.
-     *
-     * @param {Request} request
-     * @param {Error} error
-     *
-     * @private
-     */
-    _assignError(request, error) {
-        const deferred = this._displayNames[request.resModel][request.args[0]].deferred;
-        deferred.reject(error);
-        this._displayNames[request.resModel][request.args[0]] = {
-            state: "ERROR",
-            deferred,
-            error,
-        };
-    }
-
     /**
      * This method is called when the display name of a record is not in the
-     * cache. It creates a deferred and fetches the display name in the
-     * background.
+     * cache. It fetches the display name in the background.
      *
      * @param {string} model
      * @param {number} id
      *
      * @private
      * @returns {Deferred<string>}
      */
-    async _fetchDisplayName(model, id) {
-        const deferred = new Deferred();
+    _fetchDisplayName(model, id) {
         if (!this._displayNames[model]) {
             this._displayNames[model] = {};
         }
-        this._displayNames[model][id] = {
-            state: "PENDING",
-            deferred,
-        };
-        const endpoint = this._getEndpoint(model);
-        const request = new Request(model, "name_get", [id]);
-        endpoint.call(request);
-        return deferred;
+        this._displayNames[model][id] = { state: "PENDING" };
+        this._nameService.loadDisplayNames(model, [id]).then(async (displayNames) => {
+            this._displayNames[model][id].value = displayNames[id];
+            if (displayNames[id] === "string") {
+                this._displayNames[model][id].state = "COMPLETED";
+            } else {
+                this._displayNames[model][id].state = "ERROR";
+            }
+            if (this._blockNotification) {
+                return;
+            }
+            this._blockNotification = true;
+            await Promise.resolve();
```

Comment:
according to Aaron, this should not be done: in some sense, we should not take into consideration that the name service batch the requests. Another thing is that if the notification only triggers a rendering, OWL is already taking care of that.

Reply:
Instead, something could be done in the listener of "labels-fetched" or elsewhere. Group notification could be useful in other cases.

Reply:
I agree, good idea :+1:  :)